app/submitter/submitter.py

```diff
@@ -132,11 +141,13 @@ def send_message(self, message, tx_id, case_id):
 
 
 class GCSFeedbackSubmitter:
-    def __init__(self, bucket_name):
+    def __init__(self, bucket_name: str) -> None:
         client = storage.Client()
         self.bucket = client.get_bucket(bucket_name)
 
-    def upload(self, metadata, payload):
+    def upload(
+        self, metadata: Mapping[str, Union[str, int, list]], payload: dict
```

Comment:
FeedbackMetadata's `__call__` return type is `dict[str, str]`, so this should be `Mapping[str, str]`?

If Lewis' PR merges first, the type for `payload` will change to a `str`.

Reply:
Changed to `Mapping[str, str]`, keeping payload:dict for now as Lewis's PR didn't merge yet.